pkg/remediator/crashloopbackoffrescheduler_test.go

```diff
@@ -0,0 +1,148 @@
+package remediator_test
+
+import (
+	"context"
+	"errors"
+	"github.com/aksgithub/kube_remediator/pkg/k8s/mock"
+	"github.com/aksgithub/kube_remediator/pkg/remediator"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+	"gotest.tools/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	"sync"
+	"testing"
+)
+
+type TestCrashLoopBackOffReschedulerSuite struct {
+	suite.Suite
+	logger         *zap.Logger
+	mockController *gomock.Controller
+	mockClient     *mock_k8s.MockClientInterface
+	pods           []corev1.Pod
+	t              *testing.T
+}
+
+func TestSuiteCrashLoopBackOffRescheduler(t *testing.T) {
+	suite.Run(t, &TestCrashLoopBackOffReschedulerSuite{t: &testing.T{}})
+}
+
+func (suite *TestCrashLoopBackOffReschedulerSuite) SetupSuite() {
+	suite.logger, _ = zap.NewDevelopment()
+	suite.mockController = gomock.NewController(suite.t)
+	suite.mockClient = mock_k8s.NewMockClientInterface(suite.mockController)
+}
+
+func (suite *TestCrashLoopBackOffReschedulerSuite) TearDownSuite() {
+	// suite.mockController.Finish() // TODO: I think we should call this, but it crashes
+}
+
+func (suite *TestCrashLoopBackOffReschedulerSuite) testGetRemediator() (*remediator.CrashLoopBackOffRescheduler, error) {
+	remediator.CONFIG_FILE = "../../config/crash_loop_back_off_rescheduler.json"
+	remediator, err := remediator.NewCrashLoopBackOffRescheduler(suite.logger, suite.mockClient)
+	return remediator, err
+}
+
+func (suite *TestCrashLoopBackOffReschedulerSuite) SetupTest() {
+	suite.pods = []corev1.Pod{{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
still prefer 2 pods instead of 1 which will also make sure multiple pods are handled (tests loop)

Reply:
I'll do 1 test with multiple pods to make sure it loops